staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -640,3 +661,20 @@ func (in crdStorageMap) clone() crdStorageMap {
 	}
 	return out
 }
+
+// crdConversionRESTOptionsGetter overrides the codec with one using the
+// provided custom converter and custom encoder and decoder version.
+type crdConversionRESTOptionsGetter struct {
+	generic.RESTOptionsGetter
+	converter      runtime.ObjectConvertor
+	encoderVersion schema.GroupVersion
+	decoderVersion schema.GroupVersion
+}
+
+func (t crdConversionRESTOptionsGetter) GetRESTOptions(resource schema.GroupResource) (generic.RESTOptions, error) {
+	ret, err := t.RESTOptionsGetter.GetRESTOptions(resource)
+	if err == nil {
+		ret.StorageConfig.Codec = versioning.NewCodec(ret.StorageConfig.Codec, ret.StorageConfig.Codec, t.converter, &unstructuredCreator{}, discovery.NewUnstructuredObjectTyper(nil), &unstructuredDefaulter{delegate: Scheme}, t.encoderVersion, t.decoderVersion)
```

Comment:
`discovery.NewUnstructuredObjectTyper(nil)` or `discovery.NewUnstructuredObjectTyper()`? passing nil looks like it would fall back to dereferencing a nil object here:
https://github.com/kubernetes/kubernetes/blob/bf790f4467e0d01679665fc5772830bb26a83cb6/staging/src/k8s.io/client-go/discovery/unstructured.go#L61-L62

Reply:
follow-up (stated long ago by @deads2k): move `UnstructuredObjectTyper` somewhere here.